libkbfs/dirty_bcache.go

```diff
@@ -67,44 +115,46 @@ type DirtyBlockCacheStandard struct {
 	// accepted. Used only for testing.
 	blockedChanForTesting chan<- int64
 
-	// The minimum (and initial) size of the sync buffer.
-	minSyncBufferSize int64
-	// The maximum size of the sync buffer.  Also used as the
+	// The minimum (and initial) capacity of the sync buffer.
+	minSyncBufCap int64
+	// The maximum capacity of the sync buffer.  Also used as the
 	// denominator when calculating backpressure, such that the closer
 	// we are to reaching the maximum size (over and above the current
 	// sync buffer), the more write requests will be delayed.
-	maxSyncBufferSize int64
+	maxSyncBufCap int64
 
 	shutdownLock sync.RWMutex
 	isShutdown   bool
 
-	lock               sync.RWMutex
-	cache              map[dirtyBlockID]Block
-	unsyncedDirtyBytes int64
-	syncingDirtyBytes  int64 // just for bookkeeping, not actually used
-	totalDirtyBytes    int64
-	syncBufferSize     int64
+	lock            sync.RWMutex
+	cache           map[dirtyBlockID]Block
+	syncBufBytes    int64
+	waitBufBytes    int64
+	syncBufferCap   int64
+	ignoreSyncBytes int64 // these bytes have "timed out"
+	syncStarted     time.Time
```

Comment:
There's a lot to get right here with the math, and it's probably too much to expect for someone to understand all the math +/- scenarios that can happen with syncing and waiting. I think it would be a good idea to instrument these totals and log them all as part of each operation so we can perform inflow = outflow sanity checks. This would help us find leaks if there are any.